contrib/integrations/arsenal/plugin-arsenal/main.go

```diff
@@ -73,145 +108,293 @@ const deployData = `{
-	maxRetry, err := strconv.Atoi(maxRetryStr)
+	// Read and check inputs
+	var (
+		application     = getStringOption(q, "cds.application")
+		arsenalHost     = getStringOption(q, "cds.integration.deployment.host")
+		deploymentToken = getStringOption(q, "cds.integration.deployment.deployment.token", "cds.integration.deployment.token")
+		alternative     = getStringOption(q, "cds.integration.deployment.alternative.config")
+	)
+	maxRetry, err := getIntOption(q, "cds.integration.deployment.retry.max")
 	if err != nil {
 		fmt.Printf("Error parsing cds.integration.deployment.retry.max: %v. Default value will be used\n", err)
 		maxRetry = 10
 	}
-	delayRetry, err := strconv.Atoi(delayRetryStr)
+	delayRetry, err := getIntOption(q, "cds.integration.deployment.retry.delay")
 	if err != nil {
 		fmt.Printf("Error parsing cds.integration.deployment.retry.max: %v. Default value will be used\n", err)
 		delayRetry = 5
 	}
-
-	deployData, err := interpolate.Do(deployData, q.GetOptions())
-	if err != nil {
-		return fail("Error: unable to interpolate data: %v. Please check you integration configuration\n", err)
+	if arsenalHost == "" {
+		return fail("missing arsenal host")
+	}
+	if deploymentToken == "" {
+		return fail("missing arsenal deployment token")
 	}
 
-	httpClient := cdsclient.NewHTTPClient(60*time.Second, false)
+	arsenalClient := newArsenalClient(arsenalHost, deploymentToken)
 
-	// Prepare the request
-	req, err := http.NewRequest(http.MethodPost, arsenalHost+"/deploy", strings.NewReader(deployData))
-	if err != nil {
-		return fail("Error: unable to prepare request on %s/deploy: %v", arsenalHost, err)
-	}
-	req.Header.Add("X-Arsenal-Deployment-Token", arsenalDeploymentToken)
+	// Read alternative if configured.
+	var altConfig *alternativeConfig
+	if len(alternative) > 0 {
+		// Resolve alternative.
+		altTmpl, err := template.New("alternative").Delims("[[", "]]").Funcs(interpolate.InterpolateHelperFuncs).Parse(alternative)
+		if err != nil {
+			return fail("failed to resolve alternative config: %v\n", err)
+		}
+		var altBuf bytes.Buffer
+		if err = altTmpl.Execute(&altBuf, q.GetOptions()); err != nil {
+			return fail("failed to interpolate alternative config: %v", err)
+		}
 
-	fmt.Printf("Deploying %s on Arsenal at %s...\n", application, arsenalHost)
+		// Create alternative if anything was resolved.
+		if altBuf.Len() > 0 {
+			if err = json.Unmarshal(altBuf.Bytes(), &altConfig); err != nil {
+				fmt.Println("Resolved alternative:", altBuf.String())
+				return fail("failed to unmarshal alternative config: %v", err)
+			}
 
-	// Do the request
-	res, err := httpClient.Do(req)
-	if err != nil {
-		return fail("Error: Post %s/deploy failed: %v. Please check you integration configuration", arsenalHost, err)
+			// Create alternative on /alternative
+			fmt.Printf("Creating alternative %s on Arsenal...\n", altConfig.Name)
+			if err = arsenalClient.upsertAlternative(altConfig); err != nil {
```

Comment:
For readability it could be nice to print the interpolated alternative config (json prettyfied)

Reply:
It is actually printed, but only in case of failure. Do you think it should always be printed?

Reply:
Added a print to a prettified json with enriched cds metadata.